### PR TITLE
CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by mxsdev**
This is based off of the guide from `vscode-runme`.

Probably could use a look-over by @adambabik

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+This is an outline of what the workflow for code contributions looks like
+
+- Check the list of open [issues](../../issues). Either assign an existing issue to yourself, or
+   create a new one that you would like work on and discuss your ideas and use cases.
+
+It is always best to discuss your plans beforehand, to ensure that your contribution is in line with our goals.
+
+- Fork the repository on GitHub
+- Create a topic branch from where you want to base your work. This is usually master.
+- Open a new pull request, label it `work in progress` and outline what you will be contributing
```
Comment (adambabik) [inline CONTRIBUTING.md:34]: We should add this label.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+- Make commits of logical units.
+- Make sure you sign-off on your commits `git commit -s -m "adding X to change Y"`
+- Write good commit messages (see below).
+- Push your changes to a topic branch in your fork of the repository.
+- As you push your changes, update the pull request with new infomation and tasks as you complete them
+- Project maintainers might comment on your work as you progress
+- When you are done, remove the `work in progess` label and ping the maintainers for a review
+- Your pull request must receive a :thumbsup: from two [maintainers](MAINTAINERS)
```
Comment (adambabik) [inline CONTRIBUTING.md:42]: We don't have such a file/reference.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+Note that for protocol buffers to work with `npm` projects, you'll need to add the [`@buf` registry](https://docs.buf.build/bsr/remote-packages/npm) to your `npm` or `yarn` config.
+
+[registry]: https://buf.build/gen/npm/v1/@buf/stateful_runme.community_timostamm-protobuf-ts
+
+#### GraphQL
+
+GraphQL schema are generated as part of `make generate` (`go generate`).
+
+This project uses [genqlient](https://github.com/Khan/genqlient) to generate a graphql client for interacting with the stateful API. You will need to install `genqlient` in order for `go generate` to work properly.
```
Comment (adambabik) [inline CONTRIBUTING.md:86]:
```suggestion
This project uses [genqlient](https://github.com/Khan/genqlient) to generate a GraphQL client for interacting with the Stateful API. You will need to install `genqlient` in order for `go generate` to work properly.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+See also [the readme](/internal/client/graphql/schema/README.md) for generating graphql schema from the remote endpoint, which is a pre-requisite for running `go generate`.
```
Comment (adambabik) [inline CONTRIBUTING.md:88]:
```suggestion
See also [the README](/internal/client/graphql/schema/README.md) for generating GraphQL schema from the remote endpoint, which is a pre-requisite for running `go generate`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+## Contribute code
+
+This is an outline of what the workflow for code contributions looks like
+
+- Check the list of open [issues](../../issues). Either assign an existing issue to yourself, or
+   create a new one that you would like work on and discuss your ideas and use cases.
+
+It is always best to discuss your plans beforehand, to ensure that your contribution is in line with our goals.
+
+- Fork the repository on GitHub
```
Comment (adambabik) [inline CONTRIBUTING.md:32]: Nitpick: some bullet points end with `.` and others don't.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+- Create a topic branch from where you want to base your work. This is usually master.
+- Open a new pull request, label it `work in progress` and outline what you will be contributing
+- Make commits of logical units.
+- Make sure you sign-off on your commits `git commit -s -m "adding X to change Y"`
```
Comment (adambabik) [inline CONTRIBUTING.md:36]: Maybe we can link to [the GitHub docs](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) here? This is not so obvious.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+- Write good commit messages (see below).
+- Push your changes to a topic branch in your fork of the repository.
+- As you push your changes, update the pull request with new infomation and tasks as you complete them
+- Project maintainers might comment on your work as you progress
+- When you are done, remove the `work in progess` label and ping the maintainers for a review
```
Comment (adambabik) [inline CONTRIBUTING.md:41]: This label needs to be added but IMO it's redundant because GitHub has "draft pull requests" now. I would rephrase to use them instead.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+- Your pull request must receive a :thumbsup: from two [maintainers](MAINTAINERS)
+
```
Comment (adambabik) [inline CONTRIBUTING.md:43]: We can add a paragraph about linting and testing.

**Review by adambabik — CHANGES_REQUESTED** [review]

**Review by adambabik — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (mxsdev) [reply CONTRIBUTING.md]: Replaced with draft PR status

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+There are multiple ways of getting involved:
+
+- [Report a bug](#report-a-bug)
+- [Suggest a feature](#suggest-a-feature)
+- [Contribute code](#contribute-code)
+
+## Report a bug
+
+Reporting bugs is one of the best ways to contribute. Before creating a bug report, please check that an [issue](/issues) reporting the same problem does not already exist. If there is such an issue, you may add your information as a comment.
```
Comment (christian-bromann) [inline CONTRIBUTING.md:13]:
```suggestion
Reporting bugs is one of the best ways to contribute. Before creating a bug report, please check that an [issue](../../issues) reporting the same problem does not already exist. If there is such an issue, you may add your information as a comment.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+- Check the list of open [issues](../../issues). Either assign an existing issue to yourself, or
+   create a new one that you would like work on and discuss your ideas and use cases.
+
+It is always best to discuss your plans beforehand, to ensure that your contribution is in line with our goals.
+
+- Fork the repository on GitHub
+- Create a topic branch from where you want to base your work. This is usually `main`
+- Open a new draft pull request and outline what you will be contributing
+- Make commits of logical units
+- Make sure you [sign-off on your commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) `git commit -s -m "adding X to change Y"`
```
Comment (christian-bromann) [inline CONTRIBUTING.md:36]: Is this really a requirement we want to uphold? In my experience no one has this set up (not even all of us) so wonder if it makes sense to keep it

File: CONTRIBUTING.md (hunk 9 above)
Comment (mxsdev) [reply CONTRIBUTING.md:13]: @christian-bromann 

Would the link not be top level since the file is at top level for the repo? The href is copied directly from the CONTRIBUTING file in `vscode-runme`

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+- Write good commit messages (see below)
```
Comment (christian-bromann) [inline CONTRIBUTING.md:37]: not sure to what "see below" references
```suggestion
- Write [good commit messages](https://cbea.ms/git-commit/)
```

File: CONTRIBUTING.md (hunk 10 above)
Comment (mxsdev) [reply CONTRIBUTING.md:36]: Probably not, it was just in the original CONTRIBUTING file from `vscode-runme` so I kept it 🤷 will remove

**Review by sourishkrout — CHANGES_REQUESTED** [review]
Why no command block I can run with Runme @mxsdev?

**Comment by mxsdev** [comment]
@sourishkrout See [55b5b9c](https://github.com/stateful/runme/pull/141/commits/55b5b9cc078cea7f065b6b0071071c6537ac4f32)

**Comment by sourishkrout** [comment]
Beautiful! I wonder if we could add a `Brewfile` with homebrew deps to the repo and have the first command block be `$ brew bundle --no-lock`, @mxsdev. Almost there!

**Comment by mxsdev** [comment]
@sourishkrout done!

**Review by sourishkrout — APPROVED** [review]
👍 LGTM
